### PR TITLE
tests: try to detect cluster version using REST API

### PR DESCRIPTION
**Pull request by avsej**
fallback to version from environment only if the cluster does not agree
on version or returns invalid version string

### CODE REVIEW

File: couchbase/operations/management/cluster_describe.hxx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    cluster_info info{};
+};
+
+struct cluster_describe_request {
+    using response_type = cluster_describe_response;
+    using encoded_request_type = io::http_request;
+    using encoded_response_type = io::http_response;
+    using error_context_type = error_context::http;
+
+    std::string name;
```
Comment (jrawsthorne) [inline couchbase/operations/management/cluster_describe.hxx:60]: Should name be here?

Comment (avsej) [reply couchbase/operations/management/cluster_describe.hxx:60]: thanks. copy-paste issue.

File: test/utils/server_version.hxx [hunk 2]
```diff
@@ -45,38 +45,44 @@ struct server_version {
 
     [[nodiscard]] bool is_cheshire_cat() const
     {
-        // [7.0.0, inf)
-        return major >= 7;
+        // [7.0.0, 7.1.0)
+        return major >= 7 && minor < 1;
```
Comment (jrawsthorne) [inline test/utils/server_version.hxx:49]: == 7 ?

Comment (avsej) [reply test/utils/server_version.hxx:49]: let it me >= until we get next version defined

Comment (avsej) [reply test/utils/server_version.hxx:49]: ah, right. sorry

**Review by jrawsthorne — APPROVED** [review]

**Pull request merged**